supervisor/backups/backup.py

```diff
@@ -487,7 +496,7 @@ def _open_outer_tarfile():
 
         def _close_outer_tarfile() -> int:
             """Close outer tarfile."""
-            self._outer_secure_tarfile.close()
+            cast(SecureTarFile, self._outer_secure_tarfile).close()
```

Comment:
I think I'd prefer a `RuntimeError` if `self._outer_secure_tarfile` is None similar to other places.

Reply:
That doesn't really apply here. This is a similar situation to the other one, we already ensured its not `None` in this function it just lost track of it across the executor functions. Storing the value in a variable that does not have `| None` and using that instead should fix this as well.